demisto_sdk/commands/common/hook_validations/playbook.py

```diff
@@ -291,3 +298,57 @@ def is_using_instance(self) -> bool:
                     self.is_valid = False
                     return False
         return True
+
+    def is_script_id_valid(self, skip_dependencies=False):
+        """Checks whether a script id is valid (i.e id exists in set_id)
+        Args:
+            skip_dependencies (bool):  whether should skip id set validation or not
+        Return:
+            bool. if all scripts ids of this playbook are valid.
+        """
+        is_valid = True
+        id_set = self.get_id_set_file(skip_dependencies)
+        if not id_set:
+            print_warning("Playbook script id validation cannot run, could not read id_set.json")
+            is_valid = False
+            return is_valid
+
+        all_scripts_from_id_set = id_set.get("scripts")
+        all_tasks_from_pb = self.current_file.get('tasks', {})
+        for id, task in all_tasks_from_pb.items():
+            pb_task = task.get('task', {})
+            pb_script = pb_task.get('script')
+            pb_script_name = pb_task.get('scriptName')
+            script_entry_to_check = pb_script if not None else pb_script_name
+            integration_script_flag = "|||"
+
+            if pb_script and integration_script_flag not in pb_script:
+                is_valid = any([pb_script in id_set_dict for id_set_dict in all_scripts_from_id_set])
+            elif pb_script_name and integration_script_flag not in pb_script_name:
+                is_valid = any(
+                    [pb_script_name == id_set_dict[key].get('name') for id_set_dict in all_scripts_from_id_set
+                     for key in id_set_dict])
+
+            if not is_valid:
+                error_message, error_code = Errors.invalid_script_id(script_entry_to_check)
+                if self.handle_error(error_message, error_code, file_path=self.file_path):
+                    return is_valid
+        return is_valid
+
+    def get_id_set_file(self, skip_dependencies):
```

Comment:
Is there no such existing functionality anywhere that we can borrow?
Maybe @roysagi knows?